### PR TITLE
[ fix #4610, agda-mode ] Use cl-lib for compatibility with Emacs 27

### LINKED ISSUE #4610: Support Emacs 27.1
Emacs 27.0.91 (the second pretest for what will be the 27.1) was [announced](https://lists.gnu.org/archive/html/emacs-devel/2020-04/msg01123.html). I'm creating this issue for collecting/reporting the required changes/issues.

Initially, I got the following errors:

```
$ agda-mode compile
In toplevel form:
.cabal/share/x86_64-linux-ghc-8.10.1/Agda-2.6.2/emacs-mode/annotation.el:6:1:Error: Package cl is deprecated

In toplevel form:
.cabal/share/x86_64-linux-ghc-8.10.1/Agda-2.6.2/emacs-mode/eri.el:7:1:Error: Package cl is deprecated

In toplevel form:
.cabal/share/x86_64-linux-ghc-8.10.1/Agda-2.6.2/emacs-mode/agda-input.el:22:1:Error: Package cl is deprecated

In toplevel form:
.cabal/share/x86_64-linux-ghc-8.10.1/Agda-2.6.2/emacs-mode/agda2-highlight.el:11:1:Error: Package cl is deprecated

In toplevel form:
.cabal/share/x86_64-linux-ghc-8.10.1/Agda-2.6.2/emacs-mode/agda2-mode.el:17:1:Error: Package cl is deprecated
Unable to compile the following Emacs Lisp files:
  /home/asr/.cabal/share/x86_64-linux-ghc-8.10.1/Agda-2.6.2/emacs-mode/annotation.el
  /home/asr/.cabal/share/x86_64-linux-ghc-8.10.1/Agda-2.6.2/emacs-mode/eri.el
  /home/asr/.cabal/share/x86_64-linux-ghc-8.10.1/Agda-2.6.2/emacs-mode/agda-input.el
  /home/asr/.cabal/share/x86_64-linux-ghc-8.10.1/Agda-2.6.2/emacs-mode/agda2-highlight.el
  /home/asr/.cabal/share/x86_64-linux-ghc-8.10.1/Agda-2.6.2/emacs-mode/agda2-mode.el
```

I know Agda developers using different versions of GHC, which is good. I would appreciate if you tell us which Emacs version are you using.

### PR DESCRIPTION
**Pull request by L-TChen**
I am not sure about `lexical-let`, but it compiles. Please check `adga-input.el` carefully.

### CODE REVIEW

**Comment by L-TChen** [comment]
There are other API changes. Emacs 27 compiles `agda-mode`, but an earlier version of Emacs on Travis fails due to the absence of build-in function `mapcan`. I need to pull different versions of Emacs for testing.

File: src/data/emacs-mode/agda-input.el [hunk 1]
```diff
@@ -1,4 +1,4 @@
-;;; agda-input.el --- The Agda input method
+;;; agda-input.el --- The Agda input method -*- lexical-binding: t; -*-
```
Comment (nad) [inline src/data/emacs-mode/agda-input.el]: I guess you did not intend `-*- lexical-binding: t; -*-` to be part of the file's title.

Comment (L-TChen) [reply src/data/emacs-mode/agda-input.el]: Thanks. I moved the file-local variable to the beginning of file.

**Comment by andreasabel** [comment]
@nad, can this PR be merged?

**Comment by nad** [comment]
I have not tested it, but the changes look fine to me.

**Comment by andreasabel** [comment]
Thanks, @L-TChen and @nad.

**Pull request merged**